packages/common-all/src/dnode.ts

```diff
@@ -444,11 +444,12 @@ export class NoteUtils {
       const maybeTemplate = schema.data.template;
       if (maybeTemplate) {
         // TODO: Support xvault with user prompting for this flow
-        const tempNote = NoteUtils.getNoteByFnameFromEngine({
+        // Get first valid template note. If there are multiple template matches, apply first one.
+        // This is temp until we get xvault support
+        const tempNote = NoteUtils.getNotesByFnameFromEngine({
           fname: maybeTemplate.id,
-          vault: note.vault,
           engine,
-        });
+        })[0];
```

Comment:
How are the results sorted?  Should we check whether any of the results returned have a vault that matches the current `note.vault`, and if so return that one?

Reply:
Internally notes are stored in a map so I don't believe there is an order (unless js does things differently or we use a special map). Makes sense to find one that matches current vault though.